tests/lean/1870.lean.expected.out

```diff
@@ -1,7 +1,7 @@
 1870.lean:21:2-21:35: error: type mismatch
-  congrArg (@OfNat.ofNat Nat 0) (congrArg (@Zero.toOfNat0 Nat) ?_)
+  congrArg ?_ (congrArg ?_ ?_)
 has type
-  OfNat.ofNat 0 = OfNat.ofNat 0 : Prop
+  ?_ (?_ ?_) = ?_ (?_ ?_) : Prop
 but is expected to have type
   OfNat.ofNat 0 = OfNat.ofNat 1 : Prop
```

Comment:
Hmm, in this instance it’s not really an improvement. As a user I look at this and wonder why these metavariables couldn’t just be instantiated. But maybe this example is somewhat artificial.

Maybe add the test from the issue description to see one where it’s a UX improvement?

Reply:
Yeah, this one is a bit obscure. I've added some tests.

I tried making a version that is consistent about partially assigning metavariables to mimic a half-successful isDefEq, but I think in the end that's more confusing, since you don't know what the real type is. It might be nice having a real isDefEq diagnostic function instead.